src/enhanced-textarea.jsx

```diff
@@ -73,8 +77,17 @@ const EnhancedTextarea = React.createClass({
     };
   },
 
+  handleResize(e) {
+    this._syncHeightWithShadow(undefined, e);
+  },
+
   componentDidMount() {
     this._syncHeightWithShadow();
+    window.addEventListener('resize', this.handleResize);
+  },
+
+  componentWillUnmount: function() {
+    window.removeEventListener('resize', this.handleResize);
```

Comment:
We are currently removing all the mixins from our source code. I believe that it's a nice abstraction for this use case.
@alitaheri @newoga @mbrookes Are you fine with using it in this PR then migrate the source code to use it everywhere it's needed?